### ISSUE TITLE
Black RenderWindow when window.create is called twice on macOS

### ISSUE THREAD

**Issue by KRMisha** [SFML/SFML#1424]
On macOS, if you create an sf::RenderWindow with a constructor, and then call window.create() again, the window will **sometimes** start up fully black when the application is started.

Sample code:
```
sf::RenderWindow window(sf::VideoMode(800, 600), "SFML window", sf::Style::Default);
window.create(sf::VideoMode::getFullscreenModes()[1], "new", sf::Style::Default);
```

OS: macOS Sierra 10.12.6
Xcode 9

**Comment by mantognini** [comment]
I'm not sure what should actually be expected of such program. Is it really an issue in the first place?

**Comment by LaurentGomila** [comment]
What does "fully black" mean? Is it the client area of the window, or the decorations too? And what does it show when it's not black?

**Comment by KRMisha** [comment]
My small example might seem like something odd to do, but in practice the need could arise to instantiate a window first in a constructor initialization list, but then recreate it if a certain settings file is present or modified, to change context settings for example. Regardless, the docs say that the window should be destroyed and then created by the underlying code.

As for what I mean by fully black, the program otherwise works fine, with sound and other implemented code working (even using window events), the titlebar buttons are responsive, but the window itself does not have anything on it. Essentially it seems like it doesn't draw anything.

**Comment by LaurentGomila** [comment]
> the window itself does not have anything on it. Essentially it seems like it doesn't draw anything.

Please provide a complete and minimal example that demonstrates your problem. The 2 lines of code that you posted above don't draw anything, so the behaviour matches what you describe so far.

**Comment by KRMisha** [comment]
I used the "SFML App" template, and added the window.create() call right after the window constructor, without touching any other parts of the code.

Here's the code, with the single added line. (Note that you need to have `ResourcePath.mm` and `ResourcePath.hpp`, as well as the relevant resources in the `Resources` folder, which are all automatically created with the template.)
```
#include <SFML/Audio.hpp>
#include <SFML/Graphics.hpp>

// Here is a small helper for you! Have a look.
#include "ResourcePath.hpp"

int main(int, char const**)
{
    // Create the main window
    sf::RenderWindow window(sf::VideoMode(800, 600), "SFML window");
    // MODIFIED LINE HERE
    window.create(sf::VideoMode::getFullscreenModes()[1], "new", sf::Style::Default);

    // Set the Icon
    sf::Image icon;
    if (!icon.loadFromFile(resourcePath() + "icon.png")) {
        return EXIT_FAILURE;
    }
    window.setIcon(icon.getSize().x, icon.getSize().y, icon.getPixelsPtr());

    // Load a sprite to display
    sf::Texture texture;
    if (!texture.loadFromFile(resourcePath() + "cute_image.jpg")) {
        return EXIT_FAILURE;
    }
    sf::Sprite sprite(texture);

    // Create a graphical text to display
    sf::Font font;
    if (!font.loadFromFile(resourcePath() + "sansation.ttf")) {
        return EXIT_FAILURE;
    }
    sf::Text text("Hello SFML", font, 50);
    text.setFillColor(sf::Color::Black);

    // Load a music to play
    sf::Music music;
    if (!music.openFromFile(resourcePath() + "nice_music.ogg")) {
        return EXIT_FAILURE;
    }

    // Play the music
    music.play();

    // Start the game loop
    while (window.isOpen())
    {
        // Process events
        sf::Event event;
        while (window.pollEvent(event))
        {
            // Close window: exit
            if (event.type == sf::Event::Closed) {
                window.close();
            }

            // Escape pressed: exit
            if (event.type == sf::Event::KeyPressed && event.key.code == sf::Keyboard::Escape) {
                window.close();
            }
        }

        // Clear screen
        window.clear();

        // Draw the sprite
        window.draw(sprite);

        // Draw the string
        window.draw(text);

        // Update the window
        window.display();
    }

    return EXIT_SUCCESS;
}
```

**Comment by KRMisha** [comment]
The issue here isn't window.clear(), but window.create(), which sometimes causes the application to not draw anything (as shown in this example, on startup).

**Comment by PoetaKodu** [comment]
You created Window twice - what would you expect? This is definitely not a bug - it is just wrong usage. You expect from library to be bulletproof? This is completely wrong. Consider the following code using standard library:

```
std::vector<int> vec{ 1, 2, 3, 4 };
vec[2048] = 300;
```

Do you think standard library will protect you from wrong usage?

**Comment by DeathRay2K** [comment]
@PoetaKodu Window.create() is a supported part of the API, and according to the documentation should work. It doesn't make sense to say that using a documented and supported function is wrong usage.
According to the [documentation](https://www.sfml-dev.org/documentation/2.4.2/classsf_1_1Window.php#acf67483dc21f08d65c8835b3889b41b2)  Window.create is supposed to close and recreate the window if it is called on a window that already exists. The fact that this new window apparently cannot be drawn to is the bug.

**Comment by eXpl0it3r** [comment]
Just to be clear again, sometimes/always (specify which one) when calling create on the render window, when entering the rendering loop (from your example) then no text or sprite will be drawn?

**Comment by KRMisha** [comment]
@eXpl0it3r Yes that's correct: when calling create on the RenderWindow, after having initially initialized it with the non-default constructor, it will **sometimes** happen that no text or sprite will be drawn (it will be all or nothing). If the RenderWindow is only default-initialized (using the default constructor, `RenderWindow()`), however, the issue never arises. It seems to be a problem related to calling window.create() after the RenderWindow has been instantiated with the **non-default** constructor (`RenderWindow(VideoMode mode, const String &title, Uint32 style=Style::Default, const ContextSettings &settings=ContextSettings())`), as shown in the example above.

**Comment by mantognini** [comment]
I tried to reproduce the issue using the code posted above but never got a black window. I'm running on macOS 10.13, not 10.12. Not sure it's the issue, though. It would be useful if someone else could repro this as well.

**Comment by KRMisha** [comment]
@mantognini I updated to macOS 10.13.4, and you're indeed right, now the issue has disappeared...  I tested again, using the exact same project and conditions I had used before I updated. So it seems to have been something specific to 10.12.6. Is this an issue we want to solve if it's only specific to 10.12 (or 10.12 or older maybe)? In any case, it would be useful if anyone currently on 10.12 (or specifically 10.12.6) could see if they can reproduce this problem.
